gprofiler/utils/perf.py

```diff
@@ -3,14 +3,80 @@
 # Licensed under the AGPL3 License. See LICENSE.md in the project root for license information.
 #
 
-from gprofiler.exceptions import CalledProcessError
+from enum import Enum
+from pathlib import Path
+from threading import Event
+from typing import List
+
+from gprofiler.exceptions import CalledProcessError, PerfNoSupportedEvent
 from gprofiler.log import get_logger_adapter
 from gprofiler.utils import run_process
-from gprofiler.utils.perf_process import perf_path
+from gprofiler.utils.perf_process import PerfProcess, perf_path
 
 logger = get_logger_adapter(__name__)
 
 
+class SUPPORTED_PERF_EVENTS(Enum):  # pylint: disable=C0103
+    """
+    order here is crucial, the first one we try and succeed - will be used.
+    keep it mind that we should always use `PERF_DEFAULT` as a first try.
+    meaning - keeping with `perf`s' default is always preferred.
+    """
+
+    PERF_DEFAULT = "default"
+    PERF_SW_CPU_CLOCK = "cpu-clock"
+    PERF_SW_TASK_CLOCK = "task-clock"
+
+    def perf_extra_args(self) -> List[str]:
+        if self == SUPPORTED_PERF_EVENTS.PERF_DEFAULT:
+            return []
+        return ["-e", self.value]
+
+
+def perf_default_event_works(tmp_dir: Path, stop_event: Event) -> list:
```

Comment:
Please add some spacing in this function to make it clearer. For example - 
1. Blank between the docstring and code
2. blank between the try block and code in loop
3. blank between the try block and the final return line